src/main/webapp/app/overview/exercise-details/exercise-details-student-actions.component.ts

```diff
@@ -94,6 +98,23 @@ export class ExerciseDetailsStudentActionsComponent implements OnInit, OnChanges
             this.profileService.getProfileInfo().subscribe((profileInfo) => {
                 this.localVCEnabled = profileInfo.activeProfiles?.includes(PROFILE_LOCALVC);
                 this.athenaEnabled = profileInfo.activeProfiles?.includes(PROFILE_ATHENA);
+
+                // The online IDE is only available with correct SpringProfile and if it's enabled for this exercise
+                if (profileInfo.activeProfiles?.includes(PROFILE_THEIA)) {
+                    this.theiaEnabled = true;
+
+                    // Set variables now, sanitze later on
+                    this.theiaPortalURL = profileInfo.theiaPortalURL ?? '';
+
+                    // Verify that Theia's portal URL is set
+                    if (this.theiaPortalURL === '') {
+                        this.theiaEnabled = false;
+                    }
+
+                    // Future work: Verify that the course has Theia enabled
+                    // Future work: Verify that the exercise has Theia enabled
+                    // Future work: Verify that Theia has a valid blueprint to start up
+                }
```

Comment:
```suggestion
                // The online IDE is only available with correct SpringProfile and if it's enabled for this exercise
                this.theiaEnabled = profileInfo.activeProfiles?.includes(PROFILE_THEIA);
```
See other suggestion: remaining code could be part of the dedicated component.

Reply:
See above: I´d leave it for now but keep your suggestion in mind in case that the 3 "future work" aspects are more than one-liners, okay? @b-fein